src/app/apps/rup/mapa-camas/views/listado-internacion-capas/listado-internacion-capas.service.ts

```diff
@@ -31,26 +36,55 @@ export class ListadoInternacionCapasService {
         ]).pipe(
             auditTime(0),
             switchMap(([fechaIngresoDesde, fechaIngresoHasta, fechaEgresoDesde, fechaEgresoHasta]) => {
+                this.listadoInternacion = [];
                 if ((fechaIngresoDesde && fechaIngresoHasta) || fechaEgresoDesde && fechaEgresoHasta) {
                     return this.resumenHTTP.search({
                         organizacion: this.auth.organizacion.id,
                         ingreso: this.resumenHTTP.queryDateParams(fechaIngresoDesde, fechaIngresoHasta),
                         egreso: this.resumenHTTP.queryDateParams(fechaEgresoDesde, fechaEgresoHasta),
                         populate: 'idPrestacion'
-                    });
+                    }).pipe(
+                        switchMap(resumen => {
+                            if (!resumen.length) {
+                                return of([]);
+                            }
+                            resumen.forEach(int => {
+                                this.nuevoListado = {};
+                                this.nuevoListado['id'] = int.id;
+                                this.nuevoListado['fechaIngreso'] = int.fechaIngreso;
+                                this.nuevoListado['fechaEgreso'] = int.fechaEgreso;
+                                this.nuevoListado['paciente'] = int.paciente;
+                                this.nuevoListado['idPrestacion'] = int.idPrestacion;
+                                this.nuevoListado['diagnostico'] = int.registros.find(r => r.esDiagnosticoPrincipal && r.tipo === 'valoracion-inicial')?.concepto;
+                                this.listadoInternacion.push(this.nuevoListado);
+                            });
+                            const request = resumen.map(i =>
+                                this.camasHTTP.historialInternacion('internacion', 'medica', fechaIngresoDesde, moment().toDate(), i.id),
+                            );
+                            return forkJoin(request);
+                        })
+                    );
                 }
+                return of([]);
             }),
             map(internaciones => {
-                internaciones.forEach(internacion => {
-                    if (internacion.registros) {
-                        (internacion as any).diagnostico =
-                            internacion.registros.find(r => r.esDiagnosticoPrincipal && r.tipo === 'valoracion-inicial')?.concepto;
+                let index = 0;
+                internaciones.forEach(int => {
```

Comment:
Funciona bien pero es un poco confuso el código (Dejo sugerencia). También agregaría un pequeño comentario aclarando la intención del bloque!
(El foreach tiene incorporada la opción de un indice en sus parametros)
```suggestion
               internaciones.forEach((movimientos, index) => {
                    if (movimientos.length) {
                        const element = this.listadoInternacion[index];
                        const egreso = movimientos.find(mov => mov.extras?.egreso);
                        element['unidadOrganizativa'] = egreso?.unidadOrganizativa || movimientos[movimientos.length - 1].unidadOrganizativa;
                        this.listadoInternacion[index] = element;
                    }
                });
```